projects/miopen/src/kernels/MIOpenConvFFT.cpp

```diff
@@ -0,0 +1,4718 @@
+    R4 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 4 * 12)];
+    R5 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 5 * 12)];
+    R6 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 6 * 12)];
+
+    __syncthreads();
+
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 0 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R0;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 1 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R1;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 2 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R2;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 3 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R3;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 4 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R4;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 5 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R5;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 6 * 12) * (CFF_CHANNELS * CFF_BATCH + 64))] = R6;
+}
+
+__forceinline__ __device__ void FwdPass4_WE(unsigned int me,
+                                            unsigned int inOffset,
+                                            unsigned int outOffset,
+                                            float2 const* bufIn,
+                                            float2* bufOut)
+{
+    float2 R0, R1, R2, R3, R4, R5, R6;
+
+    R0 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 0 * 12)];
+    R1 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 1 * 12)];
+    R2 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 2 * 12)];
+    R3 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 3 * 12)];
+    R4 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 4 * 12)];
+    R5 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 5 * 12)];
+    R6 = bufIn[inOffset + ((me % 16) * 84 + (me / 16) + 6 * 12)];
+
+    __syncthreads();
+
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 0 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R0;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 1 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R1;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 2 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R2;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 3 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R3;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 4 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R4;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 5 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R5;
+    bufOut[outOffset + ((me % 16) + ((me / 16) + 6 * 12) * (CFF_CHANNELS * CFF_NFILTER + 64))] = R6;
+}
+
+extern "C" __global__
+    __launch_bounds__(192) void MIOpenConvFFT_fwd_in(const float* __restrict__ gbIn,
+                                                     float2* __restrict__ gbOut)
+{
+    unsigned int me    = threadIdx.x;
+    unsigned int batch = blockIdx.x;
+
+    __shared__ float2 lds[1344];
+
+    const float* lwbIn;
+    float2* lwbOut;
+
+    lwbIn  = gbIn + batch * CFF_IMG_W * CFF_IMG_H * 16;
+    lwbOut = gbOut + CFF_HALFW + batch * 16;
```

Comment:
I know the previous code was already fond of splitting variable definitions and declarations, but when they end up on adjacent lines I think it looks cleaner to merge them.
```suggestion
    const float* lwbIn  = gbIn + batch * CFF_IMG_W * CFF_IMG_H * 16;
    float2* lwbOut = gbOut + CFF_HALFW + batch * 16;
```
Same for following kernels